mlir/include/mlir/Dialect/MIOpenOps/LowerMIOpenOps.h

```diff
@@ -743,6 +797,9 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
     } else if (convOpType == miopen::ConvOpType::Conv2DOpType) {
```

Comment:
I'd expect update on output attributes as well. In both forward and bwd pass, output are "reshaped" to [k, NoHoWo]. However, in bww, output are "reshaped" to [NoHoWo, k]. The two attributes needs to be swapped.

Regarding on how to swap the two attributes, you could refer to how the attributes are swapped for filter tensor between forward and bwd pass.

Reply:
Fixed. https://github.com/whchung/llvm-project/blob/94e8eaf1e3b65addfc3b6aaa4b7b251e882d2c14/mlir/include/mlir/Dialect/MIOpenOps/LowerMIOpenOps.h#L689-L721

Reply:
I see your branch in working through this condition. I have a minor additional ask. Could we try to avoid duplication in assembling up the attributes? I mean, between forward and bww, the only difference is that some attributes are swapped. Could you assemble the attributes before the branch, so that both branch can share the assembled attributes. Something like [here](https://github.com/whchung/llvm-project/blob/94e8eaf1e3b65addfc3b6aaa4b7b251e882d2c14/mlir/include/mlir/Dialect/MIOpenOps/LowerMIOpenOps.h#L111-L141).